### PR TITLE
KAFKA-13414: Replace Powermock/EasyMock by Mockito in connect.storage

### PR DESCRIPTION
**Pull request by mimaison**
I've skipped the following classes as they use powermock to stub/access private and static fields/methods:
- KafkaConfigBackingStoreTest
- KafkaOffsetBackingStoreTest

Those will require some refactoring and will be updated in a separate PR.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java [hunk 1]
```diff
@@ -77,21 +74,17 @@ public void teardown() {
...
         Callback<Void> setCallback = expectSuccessfulSetCallback();
-        PowerMock.replayAll();
 
         store.set(firstSet, setCallback).get();
 
         Map<ByteBuffer, ByteBuffer> values = store.get(Arrays.asList(buffer("key"), buffer("bad"))).get();
         assertEquals(buffer("value"), values.get(buffer("key")));
         assertNull(values.get(buffer("bad")));
-
-        PowerMock.verifyAll();
```
Comment (dengziming) [inline connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java:88]: The following comments also apply for other changes:
should we call `Mockito.verify(setCallback).onCompletion()` here?

**Review by dengziming — COMMENTED** [review]
Thanks for these updates, left one minor comments, mostly LGTM.

File: connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java (hunk 1 above)
Comment (mimaison) [reply connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java:88]: Yes good point! I've pushed an update to address this. Thanks

**Comment by mimaison** [comment]
@chia7712 @tombentley Can you take a look? Thanks

File: connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java (hunk 1 above)
Comment (dengziming) [reply connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java:88]: The update LGTM.

**Review by dengziming — APPROVED** [review]

File: connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaStatusBackingStoreFormatTest.java [hunk 2]
```diff
@@ -185,82 +181,64 @@ public void deleteTopicStatus() {
...
+
         store.put(topicStatus);
 
         // check capture state
-        assertEquals(topicStatus, store.parseTopicStatus(valueCapture.getValue()));
+        assertEquals(topicStatus, store.parseTopicStatus(valueCaptor.getValue()));
         // state is not visible until read back from the log
         assertNull(store.getTopic(FOO_CONNECTOR, FOO_TOPIC));
-
-        verifyAll();
```
Comment (tombentley) [inline connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaStatusBackingStoreFormatTest.java:237]: I think the assertion on 219 would pass even if the 1st mocked interaction never happened. Do we need something to tighten up the expected behaviour? Maybe something like:

```java
verify(kafkaBasedLog, times(2)).send(any(), any(), any());
```

File: connect/runtime/src/test/java/org/apache/kafka/connect/storage/OffsetStorageWriterTest.java [hunk 3]
```diff
@@ -144,74 +126,60 @@ public void testFlushFailureReplacesOffsets() throws Exception {
...
-        final Callback<Void> callback = PowerMock.createMock(Callback.class);
+        final Callback<Void> callback = mock(Callback.class);
         // First time the write fails
         expectStore(OFFSET_KEY, OFFSET_KEY_SERIALIZED, OFFSET_VALUE, OFFSET_VALUE_SERIALIZED, callback, true, null);
-        // Second time it succeeds
-        expectStore(OFFSET_KEY, OFFSET_KEY_SERIALIZED, OFFSET_VALUE, OFFSET_VALUE_SERIALIZED, callback, false, null);
-        // Third time it has no data to flush so we won't get past beginFlush()
 
-        PowerMock.replayAll();
+        // Third time it has no data to flush so we won't get past beginFlush()
```
Comment (tombentley) [inline connect/runtime/src/test/java/org/apache/kafka/connect/storage/OffsetStorageWriterTest.java]: Now that the `// Second time it succeeds` comment has been removed it's a little be weird to just have comments about the first and third time followed by one about `// Second time it succeeds`.

**Review by tombentley — COMMENTED** [review]
A couple of nits.

**Comment by mimaison** [comment]
Thanks @tombentley, I've pushed an update.

**Review by tombentley — APPROVED** [review]
LGTM

**Pull request merged**